modules/core/core-scheduler/src/main/java/com/enonic/xp/impl/scheduler/distributed/SchedulableTaskImpl.java

```diff
@@ -93,26 +91,32 @@ public void run()
 
     private Context taskContext()
     {
-        Context taskContext = ContextAccessor.current();
-
         if ( job.getUser() == null )
         {
-            return taskContext;
+            return ContextBuilder.from( ContextAccessor.current() ).
+                authInfo( AuthenticationInfo.create().
+                    principals( RoleKeys.AUTHENTICATED, RoleKeys.EVERYONE ).
+                    user( User.ANONYMOUS ).
+                    build() ).
+                build();
+        }
+
+        final AuthenticationInfo authInfo = OsgiSupport.withService( SecurityService.class, securityService -> {
+            final VerifiedUsernameAuthToken token = new VerifiedUsernameAuthToken();
+            token.setIdProvider( job.getUser().getIdProviderKey() );
+            token.setUsername( job.getUser().getId() );
+
+            return securityService.authenticate( token );
+        } );
+
+        if ( !authInfo.isAuthenticated() )
```

Comment:
Check only if user is specified

Reply:
There is a check "if ( job.getUser() == null )" above

Reply:
second thought: this is not Scheduler responsibility to verify identity, remove this check altogether (but keep `securityService.authenticate( token );`